File: CodingTestPlatform/BOJ/1_Bronze/2/10809/10809.py
# 알파벳 찾기
# https://www.acmicpc.net/problem/10809

import sys
input = sys.stdin.readline

S = input().rstrip()
A = [-1] * 26

for i in range(97, 123):
    A[i - 97] = S.find(chr(i))

for j in A:
    print(j, end=" ")

File: CodingTestPlatform/BOJ/1_Bronze/2/10809/10809.js
/**
 * 알파벳 찾기
 * Bronze2
 * https://www.acmicpc.net/problem/10809
 */

const S = require('fs').readFileSync('/dev/stdin').toString().trim();

const num = 'z'.charCodeAt(0) - 'a'.charCodeAt(0) + 1;
const result = new Array(num).fill(-1);

S.split("").forEach((char, idx) => {
    const resultIdx = char.charCodeAt(0) - 'a'.charCodeAt(0);
    if (result[resultIdx] === -1) {
        result[resultIdx] = idx;
    }
});

console.log(result.join(" "));